### PR TITLE
Add TSA disable switch

### PR DESCRIPTION
**Pull request by MyUmmaGumma**
Add a config to disable TSA

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -297,6 +300,11 @@ func NewConfig() (*Config, []cli.Flag) {
 			Name:  "noop-uploaders",
 			Value: &cfg.NoopUploaders,
 		},
+		cli.BoolFlag{
+			Name:        "disable-tsa-titus-agent",
+			EnvVar:      "DISABLE_TSA_TITUS_AGENT",
```
[the remark this answers is not in the document]
Comment (solarkennedy) [reply config/config.go:305]: Please do `ENABLE_TRANSITION_NETWORK_MODE`

File: executor/runtime/types/types.go [hunk 2]
```diff
@@ -375,14 +375,19 @@ func GenerateV1TestPod(taskID string, resources *Resources, cfg *config.Config)
...
+//         network mode, so return Ipv6AndIpv4 instead
+func computeEffectiveNetworkMode(originalNetworkMode string, assignIPv6Address bool, disableTsa bool) string {
 	if originalNetworkMode == titus.NetworkConfiguration_UnknownNetworkMode.String() {
 		if assignIPv6Address {
 			return titus.NetworkConfiguration_Ipv6AndIpv4.String()
 		}
 		return titus.NetworkConfiguration_Ipv4Only.String()
 	}
 	if originalNetworkMode == titus.NetworkConfiguration_HighScale.String() {
+		if disableTsa {
```
Comment (solarkennedy) [inline executor/runtime/types/types.go:388]: This is not enough, you also need to catch the case where the originalNetworkMode was the transition mode.

If the transition mode is not enabled, then return dual stack

**Review by solarkennedy — CHANGES_REQUESTED** [review]

File: executor/runtime/types/types.go (hunk 2 above)
Comment (MyUmmaGumma) [reply executor/runtime/types/types.go]: Done.

File: config/config.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (MyUmmaGumma) [reply config/config.go]: Done.

**Review by solarkennedy — APPROVED** [review]

**Pull request merged**